### PR TITLE
remove sync.Pool in magicsock udp write path

### PR DESCRIPTION
**Pull request by jwhited**
Signed-off-by: Jordan Whited <jordan@tailscale.com>

### CODE REVIEW

File: wgengine/magicsock/magicsock.go [hunk 1]
```diff
@@ -3074,6 +3071,26 @@ func (c *RebindingUDPConn) closeLocked() error {
 	return c.pconn.Close()
 }
 
+func (c *RebindingUDPConn) WriteToUDPAddrPort(b []byte, addr netip.AddrPort) (int, error) {
+	for {
+		c.mu.Lock()
+		pconn := c.pconn
+		c.mu.Unlock()
```
Comment (bradfitz) [inline wgengine/magicsock/magicsock.go:3078]: future performance angle: can we replace this with an atomic.Value?

**Review by bradfitz — COMMENTED** [review]
See http://go/commit for commit message format

Like:

     wgengine/magicsock: remove sync.Pool in UDP write path, use netip

**Comment by bradfitz** [comment]
Sorry, https://github.com/tailscale/tailscale/pull/5165 ended up consuming this PR because it was all so infectious.

**Comment by jwhited** [comment]
closing in favor of #5165

**Pull request closed without merging**